### PR TITLE
New where syntax in misc places

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

**Review by KristofferC — APPROVED** [review]

File: base/checked.jl [hunk 1]
```diff
@@ -190,20 +190,20 @@ checked_add{T}(x1::T, x2::T, x3::T, x4::T, x5::T, x6::T, x7::T, x8::T) =
 Calculates `r = x-y`, with the flag `f` indicating whether overflow has occurred.
 """
 function sub_with_overflow end
-sub_with_overflow{T<:SignedInt}(x::T, y::T)   = checked_ssub_int(x, y)
-sub_with_overflow{T<:UnsignedInt}(x::T, y::T) = checked_usub_int(x, y)
+sub_with_overflow(x::T, y::T) where {T<:SignedInt}   = checked_ssub_int(x, y)
+sub_with_overflow(x::T, y::T) where {T<:UnsignedInt} = checked_usub_int(x, y)
 sub_with_overflow(x::Bool, y::Bool)           = x-y, false
```
Comment (tkelman) [inline base/checked.jl]: no longer aligned

File: base/checked.jl [hunk 2]
```diff
@@ -230,27 +230,27 @@ end
 Calculates `r = x*y`, with the flag `f` indicating whether overflow has occurred.
 """
 function mul_with_overflow end
-mul_with_overflow{T<:SignedInt}(x::T, y::T)   = checked_smul_int(x, y)
-mul_with_overflow{T<:UnsignedInt}(x::T, y::T) = checked_umul_int(x, y)
+mul_with_overflow(x::T, y::T) where {T<:SignedInt}   = checked_smul_int(x, y)
+mul_with_overflow(x::T, y::T) where {T<:UnsignedInt} = checked_umul_int(x, y)
 mul_with_overflow(x::Bool, y::Bool)           = x*y, false
```
Comment (tkelman) [inline base/checked.jl]: same here

File: base/checked.jl [hunk 3]
```diff
@@ -131,20 +131,20 @@ checked_abs(x::Bool) = x
 Calculates `r = x+y`, with the flag `f` indicating whether overflow has occurred.
 """
 function add_with_overflow end
-add_with_overflow{T<:SignedInt}(x::T, y::T)   = checked_sadd_int(x, y)
-add_with_overflow{T<:UnsignedInt}(x::T, y::T) = checked_uadd_int(x, y)
-add_with_overflow(x::Bool, y::Bool)           = x+y, false
+add_with_overflow(x::T, y::T) where {T<:SignedInt}   = checked_sadd_int(x, y)
+add_with_overflow(x::T, y::T) where {T<:UnsignedInt} = checked_uadd_int(x, y)
+add_with_overflow(x::Bool, y::Bool) = (x+y, false)
```
Comment (Sacha0) [inline base/checked.jl]: Alignment here as well?

Comment (musm) [inline base/checked.jl]: fixed

File: base/checked.jl (hunk 2 above)
Comment (musm) [inline base/checked.jl]: fixed

File: base/checked.jl (hunk 1 above)
Comment (musm) [inline base/checked.jl]: fixed

**Comment by musm** [comment]
tests pass locally

File: base/checked.jl (hunk 3 above)
Comment (tkelman) [inline base/checked.jl]: this line has less in common with the two above so I didn't mind this version, would make sense to use the same style for each _with_overflow case of this though

**Review by Sacha0 — APPROVED** [review]

File: base/checked.jl (hunk 3 above)
Comment (musm) [inline base/checked.jl]: that's what I figured... I guess I will change them all back to no tusing extra spaccing for the Bool cases

**Comment by musm** [comment]
g2g?

**Review by Sacha0 — APPROVED** [review]
Thanks @musm!

**Pull request merged**